src/modules/45-dashboards/pages/dashboardView/__tests__/DashboardViewPage.test.tsx

```diff
@@ -0,0 +1,92 @@
+import DashboardsPage from '../DashboardViewPage'
+
+const defaultLicenseObj: useLicenseStore.LicenseStoreContextProps = {
+  versionMap: {},
+  CI_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  FF_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CCM_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CD_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CHAOS_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  STO_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CV_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CET_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  SEI_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  updateLicenseStore: jest.fn(),
+  licenseInformation: {}
+}
+
+const renderComponent = ({ children }: React.PropsWithChildren<unknown> = {}): RenderResult =>
+  render(
+    <TestWrapper>
+      <DashboardsPage>{children}</DashboardsPage>
+    </TestWrapper>
+  )
+
+const generateMockSignedUrl = (mockUrl = ''): Promise<customDashboardServices.SignedUrlResponse> => {
+  return new Promise(resolve => {
+    resolve({ resource: mockUrl })
+  })
+}
+
+describe('DashboardsPage', () => {
+  const useLicenseStoreMock = jest.spyOn(useLicenseStore, 'useLicenseStore')
+  const useCreateSignedUrlMock = jest.spyOn(customDashboardServices, 'useCreateSignedUrl')
+  const useGetDashboardDetailMock = jest.spyOn(customDashboardServices, 'useGetDashboardDetail')
+  const useGetFolderDetailMock = jest.spyOn(customDashboardServices, 'useDeprecatedGetFolder')
+
+  const fetchFolderDetailMock = jest.fn()
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    jest.spyOn(customDashboardServices, 'useSearchFolders').mockImplementation(() => ({ data: {} } as any))
+    jest.spyOn(customDashboardServices, 'useGetModelTags').mockImplementation(() => ({ data: { resource: [] } } as any))
+    useGetFolderDetailMock.mockReturnValue({ data: { resource: 'folder name' }, refetch: fetchFolderDetailMock } as any)
+    useGetDashboardDetailMock.mockReturnValue({ resource: true, title: 'dashboard name' } as any)
+    useCreateSignedUrlMock.mockReturnValue({
+      mutate: generateMockSignedUrl,
+      loading: true,
+      error: null
+    } as any)
+  })
+
+  test('it should show dashboards loading animation', async () => {
+    const licenseObj: useLicenseStore.LicenseStoreContextProps = {
+      ...defaultLicenseObj,
+      licenseInformation: {
+        CD: {
+          status: LICENSE_STATE_VALUES.ACTIVE,
+          edition: Editions.ENTERPRISE
+        },
+        CE: {
+          status: LICENSE_STATE_VALUES.DELETED,
+          edition: Editions.ENTERPRISE
+        },
+        CF: {
+          status: LICENSE_STATE_VALUES.EXPIRED,
+          edition: Editions.ENTERPRISE
+        }
+      }
+    }
+    useLicenseStoreMock.mockReturnValue(licenseObj)
+
+    renderComponent()
+
+    expect(screen.queryByText('dashboards.upgrade')).not.toBeInTheDocument()
+
+    await waitFor(() => expect(screen.getByText('common.loading')).toBeInTheDocument())
```

Comment:
Nothing wrong here, but `findBy` makes it even easier
```suggestion
    expect(await screen.findByText('common.loading')).toBeInTheDocument()
```